File: app/src/main/java/com/wizeline/daggerjava/di/AppModule.java
package com.wizeline.daggerjava.di;

import com.wizeline.daggerjava.BuildConfig;
import com.wizeline.daggerjava.data.Conn;
import com.wizeline.daggerjava.data.User;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    @Singleton
    @Provides
    Conn provideConn() {
        return new Conn(BuildConfig.BASE_URL);
    }

    @Provides
    User provideUser() {
        return new User("Yisus");
    }
}

File: app/src/main/java/com/wizeline/dagger/di/AppModule.kt
package com.wizeline.dagger.di

import android.app.Application
import android.content.Context
import com.wizeline.dagger.BuildConfig
import com.wizeline.dagger.data.Conn
import dagger.Module
import dagger.Provides
import javax.inject.Scope
import javax.inject.Singleton

@Module
class AppModule {

    @Singleton
    @Provides
    fun provideConn() = Conn("")

    @Provides
    fun provideContext(application: Application) = RoomDatabase.build(application, "", 1)

}
